hours/tests/test_resource_periods_as_text.py

```diff
@@ -586,6 +628,97 @@ def test_date_period_as_text_with_time_span_resource_state_no_opening_hours(reso
     )
 
 
+@pytest.mark.django_db
+def test_resource_date_periods_as_text_ordering(resource):
+    """Test that the DatePeriods are ordered correctly in the cotent."""
+    order = [
+        ("2", None, None),
+        ("1", None, None),  # Created before 2
+        ("3", None, datetime.date(year=2023, month=1, day=1)),
+        ("4", datetime.date(year=2021, month=1, day=1), None),
+        (
+            "5",
+            datetime.date(year=2021, month=1, day=1),
+            datetime.date(year=2022, month=12, day=31),
+        ),
+        (
+            "6",
+            datetime.date(year=2021, month=1, day=1),
+            datetime.date(year=2023, month=1, day=1),
+        ),
+        (
+            "7",
+            datetime.date(year=2021, month=1, day=2),
+            datetime.date(year=2023, month=1, day=1),
+        ),
+    ]
+
+    for name, start_date, end_date in reversed(order):
+        # Create in reverse order to check that logic orders them correctly.
```

Comment:
Could either do this when setting `order`:
```py
order = reversed(...)
```
or just have the data in `order` in the wrong order to begin with:
```py
order = [
    ("7", ...),
    ...
    ("1", ...),
]
```